spring-cloud-contract-stub-runner/src/main/java/org/springframework/cloud/contract/stubrunner/StubRunnerFactory.java

```diff
@@ -106,6 +111,39 @@ private Path resolvePath(File unpackedLocation) {
 		return path;
 	}
 
+	private void removeCurrentMappings(Path path) {
+		HttpServerStub httpServerStub = HttpServerStubFactory.createServerStub();
```

Comment:
Ok, I think you misunderstood my suggestion. I suggested using the `HttpServerStub` interface like this:

```java
List<HttpServerStub> httpServerStubs = SpringFactoriesLoader.loadFactories(HttpServerStub.class, null);
if (httpServerStubs.isEmpty()) {
  httpServerStubs.add(new WireMockHttpServerStub());
}

...
					if (httpServerStubs.stream().anyMatch(h -> h.isAccepted(file.toFile())) {
						if (log.isDebugEnabled()) {
							log.debug("Deleting file [" + file.toString()
									+ "] since it contains a valid mapping.");
						}

						try {
							Files.delete(file);
						}
						catch (IOException ex) {
							log.warn("Failed to delete file [" + file.toString() + "]",
									ex);
						}
					}

```

something like that... That way we'll reuse whatever the user has provided.

Reply:
Thanks for the input :)

Reply:
Modified accordingly.